src/responder/common/negcache.h

```diff
@@ -71,7 +72,8 @@ int sss_ncache_set_uid(struct sss_nc_ctx *ctx, bool permanent,
                        struct sss_domain_info *dom, uid_t uid);
 int sss_ncache_set_gid(struct sss_nc_ctx *ctx, bool permanent,
                        struct sss_domain_info *dom, gid_t gid);
-int sss_ncache_set_sid(struct sss_nc_ctx *ctx, bool permanent, const char *sid);
+int sss_ncache_set_sid(struct sss_nc_ctx *ctx, bool permanent,
+                       struct sss_domain_info *dom, const char *sid);
```

Comment:
Aren't SIDs unique between all AD domains?
I.e. can't we just call `cache_req_object_by_sid_global_ncache_add()` in `cache_req_object_by_sid_ncache_add()` (ignoring domain)?

Reply:
Hi,

SIDs are unique and if a SID does not exists at all just adding it to the global negative cache would be ok. But if a SID exists in one domain of a multi-domain environment we currently do a multi-domain search and in all domains which are checked before the domain the SID is coming from the SID does not exist. So if the SID is looked up a second time the domain-specific negative might help to skip the lookups in the domains before.

Adding the get_domain request in some sense solves the same problem by skipping the multi-domain search and look in the right domain directly. But I still think it is worth having the domain-specific negative cache as well.

bye,
Sumit

Reply:
Hm... Indeed, the point is that fail to lookup in a single domain would prevent discovery in other domains.
Ok, probably not worth gimmicks.
Thank you.